### DISCUSSION TITLE
Control heat pump for hot water and heating

### DISCUSSION THREAD

**Discussion by zaubara** [mampfes/ha_epex_spot#144] [General]
Hi all,

I was wondering how I could best integrate spot prices for my heat pump (in my case, Vaillant via myVaillant). I don't have solar, so all that matters is the energy price (and maybe outside temperature and recurring shower times).

I can control hot water and heating separately, and both can be either set to heat manually (which means they always try to maintain the set temperature) or time controlled (represented in HA as a weekly repeating calendar, ie. monday to friday from 5:30 to 22:00, editing the events title "Heating Water to 53.0°C on Home Domestic Hot Water 0" changes the target temperature and device).

Easiest solution in my mind is to set them to manual and use the EPEX Spot Sensor and an automation to set the temperature to the actual target temperature for the cheapest X hours and to something ridiculously low at other times.
However, I'd rather not take a cold shower - so I was wondering if there's a smarter approach.
Maybe one that uses the Vaillant-native time control feature? One that just nudges the heat pump that "now" would be a good time to heat up the water tank a bit. Energy might be cheap, outside temperature reached a day-high (which should increasing the pumps efficiency?) - stuff like that.
SG-ready devices like this one should have a feature like that, but they are probably not exposed via API.

Are there existing solutions I have missed? Any ideas I could try? Thanks!

**Comment by mampfes** [comment]
Hi, I think there is no simple solution for your specific problem. What I would recommend is to use the data provided by the EPEX Spot integration and build an own automation on top of it. This provides the maximum flexibility.
